tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/rag/RagListener.java

```diff
@@ -147,34 +151,72 @@ public Publisher<Void> reactiveEvent(Event event) {
+            .flatMap(content -> computeMetaData(addedEvent, messageResult, session)
+                .flatMap(metaData -> {
+                    return openRagHttpClient.addDocument(
                     partitionFactory.forUsername(addedEvent.getUsername()),
                     new DocumentId(messageResult.getMessageId()),
                     content,
-                    computeMetaData(addedEvent, messageResult)));
+                    metaData);
+                }));
     }
 
-    private Map<String, String> computeMetaData(MailboxEvents.Added addedEvent, MessageResult messageResult) {
+    private Mono<Map<String, String>> computeMetaData(MailboxEvents.Added addedEvent, MessageResult messageResult, MailboxSession session) {
         try {
             Message mimeMessage = parseMessage(messageResult.getFullContent().getInputStream());
             String text = new MessageContentExtractor()
                 .extract(mimeMessage)
                 .getTextBody()
                 .orElse("");
-            return  Map.of("email.subject", mimeMessage.getSubject(),
-                "email.date", DateTimeFormatter.ISO_INSTANT.format(mimeMessage.getDate().toInstant()),
-                "email.threadId", messageResult.getThreadId().serialize(),
-                "email.doctype", "com.linagora.email",
-                "email.preview", Preview.compute(text).getValue());
+            String inReplyTo = mimeMessage.getHeader().getField("In-Reply-To") != null ?
+                getInReplyTo(mimeMessage) : "";
+
+            return getEmailIdFromMessageId(messageResult, session, inReplyTo)
+                .defaultIfEmpty("")
+                .map(parentId -> Map.of(
+                    "email.subject", mimeMessage.getSubject(),
+                    "datetime", DateTimeFormatter.ISO_INSTANT.format(mimeMessage.getDate().toInstant()),
+                    "parent_id", parentId,
+                    "relationship_id", messageResult.getThreadId().serialize(),
+                    "doctype", "com.linagora.email",
+                    "email.preview", Preview.compute(text).getValue()));
         } catch (Exception e) {
             throw new RuntimeException(e);
         }
     }
 
+    @SuppressWarnings("checkstyle:WhitespaceAround")
+    private Mono<String> getEmailIdFromMessageId(MessageResult messageResult, MailboxSession session, String inReplyTo) {
+        return (threadIdGuessingAlgorithm.getMessageIdsInThread(
+            messageResult.getThreadId(), session).collectList().flatMapMany(
+            messageIds -> {
+                return messageIdManager.getMessagesReactive(messageIds, FetchGroup.HEADERS, session);
+            })
+            .filter(msg -> {
+                try {
+                    Message mimeMsg = parseMessage(msg.getFullContent().getInputStream());
+                    String msgId = mimeMsg.getHeader().getField("Message-ID") == null ? "" :  mimeMsg.getHeader().getField("Message-ID").getBody().trim().replaceAll("[<>]", "");
```

Comment:
We likely can do smarter replace without relying on regexp ie do not use `replaceAll`